src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -76,7 +70,14 @@ public bool RequireHttps
             try
             {
                 IdentityModelEventSource.Logger.WriteVerbose(LogMessages.IDX10805, address);
-                HttpResponseMessage response = await _httpClient.GetAsync(address, cancel).ConfigureAwait(false);
+                HttpResponseMessage response;
+
+                if (_httpClient == null)
+                    using (HttpClient _httpClient = new HttpClient())
```

Comment:
Agree. I will update it to use a static httpClient object instead of either creating or disposing it everytime.